src/pixelator/graph/backends/implementations.py

```diff
@@ -628,157 +697,335 @@ def add_names_to_vertexes(self, vs_names: List[str]) -> None:
+        """Select a subset of edges."""
+        return IgraphBasedEdgeSequence(self._edge_seq.select(_within=component))
 
 
 class IgraphBasedVertexClustering(VertexClustering):
     """Wrapper class for a cluster of vertexes."""
 
-    def __init__(self, raw) -> None:
+    def __init__(
+        self, vertex_clustering: igraph.VertexClustering, graph: igraph.Graph
+    ) -> None:
         """Instantiate a VertexClustering."""
-        self._raw = raw
+        self._vertex_clustering = vertex_clustering
+        self._graph = graph
 
     def __len__(self) -> int:
         """Get the number of clusters."""
-        return len(self._raw)
+        return len(self._vertex_clustering)
 
-    def __iter__(self) -> Iterable[VertexSequence]:
+    def __iter__(self) -> Iterator[VertexSequence]:
         """Provide an iterator over the clusters."""
-        for cluster in self._raw:
-            yield IgraphBasedVertexSequence(cluster)
+        for cluster in self._vertex_clustering:
+            yield IgraphBasedVertexSequence(self._graph.vs[cluster], self._graph)
 
     @property
     def modularity(self) -> float:
         """Get the modularity of the clusters."""
-        return self._raw.modularity
+        return self._vertex_clustering.modularity
 
     def crossing(self) -> EdgeSequence:
-        """Get any crossing edges."""
-        return self._raw.crossing()
+        """Get crossing edges."""
+        crossing_indexes = [
+            idx
+            for idx, is_crossing in enumerate(self._vertex_clustering.crossing())
+            if is_crossing
+        ]
+        return IgraphBasedEdgeSequence(self._graph.es.select(crossing_indexes))
 
     def giant(self) -> Graph:
         """Get the largest component."""
         from pixelator.graph import Graph
 
-        return Graph(IgraphGraphBackend(self._raw.giant()))
+        return Graph(IgraphGraphBackend(self._vertex_clustering.giant()))
 
     def subgraphs(self) -> Iterable[Graph]:
         """Get subgraphs of each cluster."""
         from pixelator.graph import Graph
 
-        return [Graph(IgraphGraphBackend(g)) for g in self._raw.subgraphs()]
+        return [
+            Graph(IgraphGraphBackend(g)) for g in self._vertex_clustering.subgraphs()
+        ]
+
+
+class NetworkxBasedVertex(Vertex):
+    """A Vertex instance that plays well with NetworkX."""
+
+    def __init__(self, index: int, data: Dict):
+        """Create a new NetworkxBasedVertex instance."""
+        self._index = index
+        self._data = data
+
+    @property
+    def index(self):
+        """Get the index of the vertex."""
+        return self._index
+
+    @property
+    def data(self) -> Dict:
+        """Get the of the vertex as a dict."""
+        return self._data
+
+    def __getitem__(self, attr: str) -> Any:
+        """Get the attr of the provided vertex."""
+        return self._data[attr]
+
+    def __setitem__(self, attr: str, value: Any) -> None:
```

Comment:
Should be easy to test to cover this, right?

Reply:
Good catch. This is fixed now.